Login System

sharmaadvay0 opened the pull request:
This pull request implements the entire login system and puts the Average Chart on the home page.

Commit:
initial login system

Commit:
second login system

Commit:
third login commit

sharmaadvay0 committed:
login system completed

sharmaadvay0 committed:
removed unused imports

sharmaadvay0 committed:
removed unused logs and made average chart conditionally render if there is a token

timstjiang reviewed (commented):
I checked out this commit and tried to log in, got this error. Not sure if this is a code issue or an issue with different clients.
![image](https://user-images.githubusercontent.com/46114553/103131029-f9ac9880-4653-11eb-8cac-904efa1b618c.png)

sharmaadvay0 committed:
added alert for incorrect email or password

sharmaadvay0 commented:
Fixed the bug where the page would crash if the login was incorrect. Added an alert for an incorrect login:
![Screen Shot 2020-12-25 at 11 41 23 AM](https://user-images.githubusercontent.com/52980112/103141945-fb587980-46b0-11eb-954c-a2845d7bd124.png)

Samir-Rashid reviewed (approved):
The authentication looks good to me. It all works correctly.
